src/init.cpp

```diff
@@ -1276,6 +1276,30 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
                                      chainman, *node.mempool, ignores_incoming_txs);
     RegisterValidationInterface(node.peerman.get());
 
+    // Check port numbers
+    const bilingual_str port_error = _("Invalid port specified in %s: %s");
+    const std::string port_options_int[] = {"-port", "-rpcport"};
+    for (const std::string& port_option : port_options_int) {
+        if (args.IsArgSet(port_option)) {
+            int64_t portIn = args.GetIntArg(port_option, 0);
+            if (portIn <= 0 || portIn > std::numeric_limits<uint16_t>::max())
+                return InitError(strprintf(port_error, port_option, ToString(portIn)));
+        }
+    }
+
+    std::string hostOut;
+    uint16_t portOut{0};
+    const std::string port_options_str[] = {"-bind", "-i2psam", "-onion", "-proxy", "-rpcbind",
+      "-torcontrol", "-whitebind", "-zmqpubhashblock","-zmqpubhashtx","-zmqpubrawblock", "-zmqpubsequence"};
+    for (const std::string& port_option : port_options_str) {
+        for (std::string socketAddr : args.GetArgs(port_option)) {
+            if (port_option == "-bind")
+                socketAddr = socketAddr.substr(0, socketAddr.rfind('='));
```

Comment:
Include braces, even if it's just a one line body.

```suggestion
            if (port_option == "-bind") {
                socketAddr = socketAddr.substr(0, socketAddr.rfind('='));
            }
```